src/parser/esprima_cpp/esprima.cpp

```diff
@@ -2148,6 +2047,9 @@ class Parser {
     template <class ASTBuilder>
     ASTNode parseNewExpression(ASTBuilder& builder)
     {
+        if (this->lookahead.end - this->lookahead.start != 3) {
```

Comment:
LGTM after a minor correction
You've omitted adding comment and replacing with `hasAllocatedString` at here.

Reply:
In keyword case, hasAllocatedString is always true.(new is always keyword even escaped)
but I don't want to change `hasAllocatedString` for performance